src/preprocessing/preprocessing-modules/ImageURLRewrite.ts

```diff
@@ -0,0 +1,49 @@
+import { Config } from '../../config';
+import { parseHTML, serializeHTML } from '../util';
+
+const IMAGE_ELEMENT_SELECTOR = 'amp-img,amp-anim';
+
+/**
+ * Rewrites all images (amp-img and amp-anim) to use the proxy server from the
+ * config, if set.
+ *
+ * @param {string} amp AMP code to modify images inside
+ * @param {!Config} config Global config
+ * @return {string} Modified AMP code
+ */
+function process(amp: string, config: Config): string {
+  // Only relevant if image proxying is enabled
+  if (!config.imageProxyURL) {
+    return amp;
+  }
+
+  const doc = parseHTML(amp);
+  const images = doc.querySelectorAll(IMAGE_ELEMENT_SELECTOR);
+  for (const img of Array.from(images)) {
+    rewriteImageURL(img, config.imageProxyURL);
```

Comment:
One can use mustache templates in the src attribute of an image.  If say the image src changes, we will need to be able to rewrite them accordingly.

    e.g.
     <button on="tap:myList.refresh">Refresh List</button>
     <amp-list height="100" width="100" id="myList" src="https://some-list.com/asd.json">
        <amp-img height="100" width="100" src="https:{src}"></amp-img>
      </amp-list>

Reply:
we need to make that mustache example visible somewhere if it isn't. I'll take a look.

Reply:
This is a really good point! Totally forgot about that.

I added a regex to skip these and also added a test. It turned out that `querySelectorAll` actually ignores templates, so this was working even without the regex, but I kept it just in case. Let me know what you think.

This also means that this module doesn't rewrite image URLs inside those, but that's also WAI I think - those need to be handled by the XHR proxy anyway and this module wouldn't catch `<script>` based templates anyway.

Reply:
https://ggdmfw.com/playground/#amp-list  amp-list #2 example actually has an example though there is poor visibility on what is being showcased in the example. I think comments in the markup itself or somewhere would be super useful. I'll create a tracking bug to fix this internally.